akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala

```diff
@@ -43,7 +43,16 @@ private[akka] object ActorContextAdapter {
 
   private[akka] override def currentBehavior: Behavior[T] = adapter.currentBehavior
 
-  final override val self = ActorRefAdapter(classicContext.self)
+  // optimization to avoid adapter allocation unless used
+  // self documented as thread safe, on purpose not volatile since
+  // lazily created because creating adapter is idempotent
+  private var _self: OptionVal[ActorRef[T]] = OptionVal.None
+  override def self: ActorRef[T] = {
+    if (_self.isEmpty) {
+      _self = OptionVal.Some(ActorRefAdapter(classicContext.self))
+    }
+    _self.get
```

Comment:
could use `match` Some/None, but not important

Reply:
using `if else` for performance?

Reply:
So, is there any reason why not using `match` here?

Reply:
One line fewer like this (not that I think that is a strong argument)